auto_submit/lib/validations/validation_filter.dart

```diff
@@ -0,0 +1,95 @@
+// Copyright 2023 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:auto_submit/configuration/repository_configuration.dart';
+import 'package:auto_submit/service/config.dart';
+import 'package:auto_submit/service/process_method.dart';
+import 'package:auto_submit/validations/approval.dart';
+import 'package:auto_submit/validations/ci_successful.dart';
+import 'package:auto_submit/validations/conflicting.dart';
+import 'package:auto_submit/validations/empty_checks.dart';
+import 'package:auto_submit/validations/revert.dart';
+import 'package:auto_submit/validations/unknown_mergeable.dart';
+import 'package:auto_submit/validations/validation.dart';
+
+/// The [ValidationFilter] allows us to pick and choose and the validations to
+/// run on a particular type of pull request.
+abstract class ValidationFilter {
+  factory ValidationFilter(
+    Config config,
+    ProcessMethod processMethod,
+    RepositoryConfiguration repositoryConfiguration,
+  ) {
+    switch (processMethod) {
+      case ProcessMethod.processAutosubmit:
+        return PullRequestValidationFilter(config, repositoryConfiguration);
+      case ProcessMethod.processRevert:
+        return RevertRequestValidationFilter(config, repositoryConfiguration);
+      default:
+        return NoOpValidationFilter(config, repositoryConfiguration);
+    }
+  }
+
+  Set<Validation> getValidations();
+}
+
+/// The [NoOpValidationFilter] allows us to return a set of validations without
+/// exploding in the event something goes wrong.
```

Comment:
This is hiding the issue. Shall we through exception explicitly instead if something goes wrong?

Reply:
There really isn't an issue. It is just the default case. I could have just as easily not included it by using if/else. If that is preferable I can change it?

Reply:
I see your point. I will throw if we get here.